dask/dataframe/multi.py

```diff
@@ -295,7 +295,7 @@ def single_partition_join(left, right, **kwargs):
         else:
             divisions = [None for _ in right.divisions]
 
-    elif right.npartitions == 1:
+    elif right.npartitions == 1 and kwargs['how'] in ('inner', 'left'):
```

Comment:
Are there cases where neither of these two branches is taken?

Reply:
These are the conditions required to enter `single_partition_join` method. 

`dask.dataframe.multi.py#370-371`

Reply:
```    elif (left.npartitions == 1 and how in ('inner', 'right') or
          right.npartitions == 1 and how in ('inner', 'left')):	          right.npartitions == 1 and how in ('inner', 'left')):
        return single_partition_join(left, right, how=how, right_on=right_on,	        return single_partition_join(left, right, how=how, right_on=right_on,
                                     left_on=left_on, left_index=left_index,	                                     left_on=left_on, left_index=left_index,
                                     right_index=right_index,	                                     right_index=right_index,
                                     suffixes=suffixes, indicator=indicator)	                                     suffixes=suffixes, indicator=indicator)
```

Reply:
Could add (if desired for future developer ease) an `else: raise NotImplementedError...` Not sure what best practices in this code base are. At the moment, the method is only called inside of that parent `elif` statement.

Reply:
Ah, ok, sounds like it's safe then.  An `else: raise` block seems reasonable to me.  I'm happy to defer to your preferences here.

Reply:
Added the raise error~